File: SectorCreator.WorldBuilder/Planet/Planet/Government.cs
using SectorCreator.WorldBuilder.Services;

namespace SectorCreator.WorldBuilder.Planet.Planet;

public class Government
{
    public string Profile => $"{Code}-{Centralization.Code}{GetAuthorityProfiles()}";
    public string FactionProfile => $"{RomanNumeralService.values[Id + 1]}-{Code}-{FactionStrength.Code}";
    public int Code { get; set; }
    public List<int> Relationships { get; set; } = new();
    public Centralization Centralization { get; set; }
    public List<GovernmentAuthority> Authorities { get; set; } = new();
    public FactionStrength FactionStrength { get; set; }
    public int LawLevel { get; set; }
    public JusticeSystem JusticeSystem { get; set; }
    public Uniformity UniformityOfLaw { get; set; }
    public bool PresumptionOfInnocence { get; set; }
    public bool DeathPenalty { get; set; }
    public int EconomicLawLevel { get; set; }
    public int CriminalLawLevel { get; set; }
    public int PrivateLawLevel { get; set; }
    public int PersonalRightsLevel { get; set; }
    public int TechLevel { get; set; }
    public int LowCommonTechLevel { get; set; }
    public int Id { get; set; }
    public int Enforcement { get; set; }
    public int Militia { get; set; }
    public int Army { get; set; }
    public int WetNavy { get; set; }
    public int AirForce { get; set; }
    public int SystemDefense { get; set; }
    public int Navy { get; set; }
    public int Marine { get; set; }
    private string GetAuthorityProfiles()
    {
        List<string> profiles = new();
        foreach (var authority in Authorities) {
            profiles.Add(authority.Profile);
        }

        return string.Join("-", profiles);
    }
}

File: TravellerCreator/ClientApp/src/app/sector-creator/models/world-builder/government.ts
import { Centralization } from "./Enums/centralization";
import {GovernmentAuthority} from "./governmentAuthority";
import {FactionStrength} from "./Enums/factionStrength";
import {JusticeSystem} from "./Enums/justiceSystem";
import {Uniformity} from "./Enums/uniformity";

export class Government {
  code: number;
  relationships: number[];
  centralization: Centralization;
  authorities: GovernmentAuthority[];
  factionStrength: FactionStrength;
  lawLevel: number;
  justiceSystem: JusticeSystem;
  uniformityOfLaw: Uniformity;
  presumptionOfInnocence: boolean;
  deathPenalty: boolean;
  economicLawLevel: number;
  criminalLawLevel: number;
  privateLawLevel: number;
  personalRightsLevel: number;
  techLevel: number;
  lowCommonTechLevel: number;
  id: number;
  enforcement: number;
  militia: number;
  army: number;
  wetNavy: number;
  airForce: number;
  systemDefense: number;
  navy: number;
  marine: number;
  profile: string;
  factionProfile: string;
}
